runtime/autoload/man.vim

```diff
@@ -57,30 +53,70 @@ function! man#open_page(count, count1, mods, ...) abort
     noautocmd execute 'silent' a:mods 'split' fnameescape(bufname)
   endif
   let b:man_sect = sect
-  call s:read_page(path)
+  call s:put_page(page)
 endfunction
 
 function! man#read_page(ref) abort
   try
     let [sect, name] = man#extract_sect_and_name_ref(a:ref)
     let [b:man_sect, name, path] = s:verify_exists(sect, name)
+    let page = s:get_page(path)
   catch
     " call to s:error() is unnecessary
     return
   endtry
-  call s:read_page(path)
+  call s:put_page(page)
+endfunction
+
+" Handler for s:system() function.
+function! s:system_handler(jobid, data, event) abort
+  if a:event == 'stdout'
+    let self.stdout .= join(a:data, "\n")
+  elseif a:event == 'stderr'
+    let self.stderr .= join(a:data, "\n")
+  else
+    let s:shell_error = a:data
+  endif
 endfunction
 
-function! s:read_page(path) abort
+" Run a system command and timeout after 30 seconds.
+function! s:system(cmd, ...) abort
+  let opts = {
+        \ 'stdout': '',
+        \ 'stderr': '',
+        \ 'on_stdout': function('s:system_handler'),
```

Comment:
I would leave it the same function. If we where doing more complex handling then sure.